### PR TITLE
feature/redshift serverless support

### PR DESCRIPTION
**Pull request by svdimchenko**
Add redshift serverless support to resolve `svv_schema_quota_state` and `svl_user_info` access restrictions.

### CODE REVIEW

**Comment by mohrman** [comment]
Would love to get this feature merged. Is there any way I can help out/push for it to get a review @svdimchenko?

**Comment by mohrman** [comment]
Hello @pawelaugustyn 
Would you be able to support with review?

**Comment by pawelaugustyn** [comment]
@winglot fyi

**Comment by mohrman** [comment]
Hi @winglot do you know the status on this review? If it's not planned in the near future I'll think alternative solutions for the issue

**Comment by olivermeyer** [comment]
Hi @robertomczak, sorry for tagging you directly but I just saw you merged some PRs recently. Are you able to help merge this one too? Would be super helpful to everyone using Redshift Serverless. Thanks :)

**Comment by robertomczak** [comment]
Hi @olivermeyer no worries. Looking briefly into this PR, I wonder if identifying if Redshift is in serverless mode can be dynamic to avoid provider user wondering what switch to use? 🧐 If only difference is what data sources has to be used, I think we can try to make it transparent. 

Other than that I think we are missing tests here in order to make sure that new and old functionalities are working as expected.

File: redshift/resource_redshift_user.go [hunk 1]
```diff
@@ -253,17 +253,31 @@ func resourceRedshiftUserRead(db *DBConnection, d *schema.ResourceData) error {
...
+		userInfoTable = "pg_user"
+	} else {
+		columns = []string{
+			"usename",
+			"usecreatedb",
+			"usesuper",
+			"syslogaccess",
+			`COALESCE(useconnlimit::TEXT, 'UNLIMITED')`,
+		}
+		userInfoTable = "svl_user_info"
```
Comment (olivermeyer) [inline redshift/resource_redshift_user.go:277]: We can avoid forking between serverless and provisioned if we use the `svv_user_info` view which is available on both instance types. The column names need to be updated to reflect the change (e.g. `usename` becomes `user_name`).

**Comment by olivermeyer** [comment]
Thanks @robertomczak. AFAIK there's no way to determine which type of Redshift instance is being used in-session. I was hoping the `version()` function would show something but it doesn't.

@svdimchenko are you still willing to help with this? I'd be happy to look into it in the new year but my knowledge of Go is almost zero so I won't be very efficient.

**Comment by winglot** [comment]
Thank you for your contribution.

> Thanks @robertomczak. AFAIK there's no way to determine which type of Redshift instance is being used in-session. I was hoping the `version()` function would show something but it doesn't.

I was thinking about simplifying the setup by getting rid of the `is_serverless` switch. After contacting AWS they suggested checking for the `SYS_SERVERLESS_USAGE` view (https://docs.aws.amazon.com/redshift/latest/dg/SYS_SERVERLESS_USAGE.html) which is only available on Serverless instances. The provider could do something like `SELECT 1 FROM SYS_SERVERLESS_USAGE LIMIT 1`, if the query goes without problems we know we run on serverless, if we get permission denied - we run on classic Redshift. The view has read permission for public so it should be available to all users on serverless.
WDYT?

Let me know if you will have time for this, otherwise, I can jump in and do the changes.

**Comment by Olaktal** [comment]
Hello
Any update about this feature ? 🙏 
Thanks

**Comment by svdimchenko** [comment]
Hi there. Sorry for long delay. I don't use Redshift anymore, so that will close the PR as I don't have possibility even to test it. Feel free to improve it if necessary.

**Pull request closed without merging**